### PR TITLE
Correct when an event is marked as past

### PR DESCRIPTION
**Pull request by samuele-mattiuzzo**
Current event: `end_date__lte = now`

Past event: `end_date__gt = now`

An event still running/ending today is a current event; any event that ended days prior to today are past.

### CODE REVIEW

**Comment by samuele-mattiuzzo** [comment]
This is a small hotfix, I should have caught this while code reviewing.

**Comment by andylolz** [comment]
Although these variables are called `date_start` and `date_end`, I think they’re actually datetimes:

https://github.com/IATI/IATI-Standard-Website/blob/b58876e8e5136fbcf7538a0f326939a31254468e/events/models.py#L91-L92

So comparisons to `now` are datetime comparisons, not date comparisons. (I’m not sure if that is helpful or relevant, but it seems like it might be. Apologies if not.)

**Comment by samuele-mattiuzzo** [comment]
> Apologies if not.

Accepted

**Comment by andylolz** [comment]
> > Apologies if not.
> 
> Accepted

Ha, thanks. Maybe I could have been clearer. I think the comments added in this PR are potentially confusing. I’ll add some comments inline to clarify what I mean.

File: events/models.py [hunk 1]
```diff
@@ -49,10 +49,10 @@ def get_context(self, request, *args, **kwargs):
         archive_years = EventPage.objects.live().descendant_of(self).filter(date_end__lte=now).dates('date_end', 'year', order='DESC')
         past = request.GET.get('past') == "1"
         if past:
-            filter_dict["date_end__lte"] = now
+            filter_dict["date_end__lt"] = now  # an event is in the past if it ended "yesterday"
             order_by = ['-date_start']
         else:
-            filter_dict["date_end__gt"] = now
+            filter_dict["date_end__gte"] = now  # an event is current / not in the past if the end date is before and up to today
```
Comment (andylolz) [inline events/models.py]: This comment is incorrect. Should be something like:

> # an event is current / not in the past if the end date is now, or in the future

File: events/models.py [hunk 2]
```diff
@@ -49,10 +49,10 @@ def get_context(self, request, *args, **kwargs):
         archive_years = EventPage.objects.live().descendant_of(self).filter(date_end__lte=now).dates('date_end', 'year', order='DESC')
         past = request.GET.get('past') == "1"
         if past:
-            filter_dict["date_end__lte"] = now
+            filter_dict["date_end__lt"] = now  # an event is in the past if it ended "yesterday"
```
Comment (andylolz) [inline events/models.py]: > # an event is in the past if it ended "yesterday"

^^ This isn’t a date comparison, it’s a _datetime_ comparison, so while the comment is true, "yesterday" here is a bit misleading. This filter gets all events that ended at least one second ago. So this end date could be today, but in the past.

**Comment by andylolz** [comment]
More generally, this PR fixes a pretty extreme edgecase (that is: if an event ends _this second_, it’ll now be considered current instead of past). I don’t know if that was the goal of the fix – a test would help clarify this.

**Pull request merged**